### PR TITLE
[MM-35811][MM-35386] Upgrade to Electron 12

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Upgrade the desktop app to Electron 12.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35811

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
@amyblais @Willyfrog Can we cherry-pick this to `release-4.7` since we haven't release it yet?

**Comment by Willyfrog** [comment]
> @amyblais @Willyfrog Can we cherry-pick this to `release-4.7` since we haven't release it yet?

if QA feels confortable testing a new electron upgrade, I'm fine with it :)

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "url": "git://github.com/mattermost/desktop.git"
   },
   "config": {
-    "target": "11.4.4",
+    "target": "12.0.6",
```
Comment (Willyfrog) [inline package.json:19]:
```suggestion
    "target": "12.0.7",
```
shouldn't this match the electron version?

**Review by Willyfrog — APPROVED** [review]
minor non-blocking edit, LGTM

**Comment by Willyfrog** [comment]
note that this might also fix: [MM-35386](https://mattermost.atlassian.net/browse(/MM-35386

**Review by jupenur — APPROVED** [review]

**Comment by amyblais** [comment]
How risky is this change?

> @amyblais @Willyfrog Can we cherry-pick this to `release-4.7` since we haven't release it yet?

@esethna @jgilliam17 Any thoughts on this?

**Comment by esethna** [comment]
@amyblais I'll leave that decision up to QA, but I would propose that QA does a thorough test pass if we are going to upgrade the electron version in a dot release

**Comment by Willyfrog** [comment]
> @amyblais I'll leave that decision up to QA, but I would propose that QA does a thorough test pass if we are going to upgrade the electron version in a dot release

@jgilliam17 ☝️

**Comment by jgilliam17** [comment]
Thanks for the ping Guillermo 
@amyblais we can add Electron 12 upgrade to 4.7 release considering it will fix the security issue and some additional window sizing issues we discussed in webplatform triage today. 
Guillermo - can I have new rc cut after this PR is merged and cherrypicked to 4.7? Thanks

**Comment by Willyfrog** [comment]
> Guillermo - can I have new rc cut after this PR is merged and cherrypicked to 4.7? Thanks

Of course!

**Review by jwilander — APPROVED** [review]

**Pull request merged**